creed/tiling_utils_test.py

```diff
@@ -0,0 +1,745 @@
+    user_inputs = iter([2, 4, 1, 2, 'N', 4, 8, 2, 4, 'y'])
+
+    # override the default functionality of the input function
+    monkeypatch.setattr('builtins.input', lambda _: next(user_inputs))
+
+    # use the dummy user data to read values into the params lists
+    tiling_utils.read_tiled_region_inputs(
+        sample_fovs_list, sample_region_params
+    )
+
+    assert (sample_region_params[i]['region_start_x'] == 100 * i
+            for i in range(len(sample_region_params)))
+    assert (sample_region_params[i]['region_start_y'] == 150 * (i + 1)
+            for i in range(len(sample_region_params)))
+    assert (sample_region_params[i]['fov_num_x'] == 2 * (i + 1)
+            for i in range(len(sample_region_params)))
+    assert (sample_region_params[i]['fov_num_y'] == 4 * (i + 1)
+            for i in range(len(sample_region_params)))
+    assert (sample_region_params[i]['x_fov_size'] == 1 * (i + 1)
+            for i in range(len(sample_region_params)))
+    assert (sample_region_params[i]['y_fov_size'] == 2 * (i + 1)
+            for i in range(len(sample_region_params)))
+
+    assert sample_region_params['region_rand'] == ['N', 'Y']
+
+
+def test_generate_region_info():
+    sample_region_inputs = {
+        'region_start_x': [100, 200],
+        'region_start_y': [200, 400],
+        'fov_num_x': [3, 6],
+        'fov_num_y': [6, 12],
+        'x_fov_size': [5, 10],
+        'y_fov_size': [10, 20],
+        'region_rand': ['N', 'Y']
+    }
+
+    # generate the region params
+    sample_region_params = tiling_utils.generate_region_info(sample_region_inputs)
+
+    # assert region_start_x set correctly
+    assert all(
+        sample_region_params[i]['region_start_x'] == 100 * (i + 1)
+        for i in range(len(sample_region_params))
+    )
+
+    # assert region_start_y set correctly
+    assert all(
+        sample_region_params[i]['region_start_y'] == 200 * (i + 1)
+        for i in range(len(sample_region_params))
+    )
+
+    # assert fov_num_x set correctly
+    assert all(
+        sample_region_params[i]['fov_num_x'] == 3 * (i + 1)
+        for i in range(len(sample_region_params))
+    )
+
+    # assert fov_num_y set correctly
+    assert all(
+        sample_region_params[i]['fov_num_y'] == 6 * (i + 1)
+        for i in range(len(sample_region_params))
+    )
+
+    # assert x_fov_size set correctly
+    assert all(
+        sample_region_params[i]['x_fov_size'] == 5 * (i + 1)
+        for i in range(len(sample_region_params))
+    )
+
+    # assert y_fov_size set correctly
+    assert all(
+        sample_region_params[i]['y_fov_size'] == 10 * (i + 1)
+        for i in range(len(sample_region_params))
+    )
+
+    # assert region_rand set correctly
+    assert sample_region_params[0]['region_rand'] == 'N'
+    assert sample_region_params[1]['region_rand'] == 'Y'
+
+
+@pytest.mark.parametrize('moly_interval_val', test_cases._PARAM_SET_MOLY_INTERVAL_VALUE_CASES)
```

Comment:
add a `parameterize = pytest.mark.paramtrize` at the top to shorten calls

Reply:
Given that this global is only used for this function, it makes more sense to just use [0, 1] here instead